compiler/src/dotty/tools/dotc/CompilationUnit.scala

```diff
@@ -30,6 +30,18 @@ class CompilationUnit protected (val source: SourceFile) {
   /** Is this the compilation unit of a Java file */
   def isJava: Boolean = source.file.name.endsWith(".java")
 
+  /** Under `-from-tasty` the attributes associated with the underlying TASTy, otherwise null.
+   * This is also null until set by ReadTasty.
+   */
+  var tastyAttributes: tasty.Attributes | Null = null
```

Comment:
It seems more consistent to have the `CompilationUnitInfo` here. Then we could reuse this instance of `CompilationInfo` in the Namer when we create a new class symbol.